File: junit/rails/game/state/ModelTest.java
package rails.game.state;

import static org.mockito.Mockito.*;
import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class ModelTest {

    private static final String MODEL_ID = "Model";
    private static final String MODEL_TEXT_INIT = "Init";
    private static final String MODEL_TEXT_CHANGE = "Change";
    
    private Root root;
    private ModelImpl model;
    @Mock private Observer observer;
    
    @Before
    public void setUp() {
        root = StateTestUtils.setUpRoot();
        model = ModelImpl.create(root, MODEL_ID, MODEL_TEXT_INIT);
        StateTestUtils.close(root);
        model.addObserver(observer);
    }
    
    @Test
    public void testModel() {
        assertEquals(MODEL_TEXT_INIT, model.toText());
        model.changeText(MODEL_TEXT_CHANGE);
        StateTestUtils.close(root);
        assertEquals(MODEL_TEXT_CHANGE, model.toText());
        verify(observer).update(MODEL_TEXT_CHANGE);
    }

}

File: dotnet/GameStateTest/ModelTest.cs
using GameLib.Net.Game.State;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GameStateTest
{
    [TestClass]
    public class ModelTest
    {
        private const string MODEL_ID = "Model";
        private const string MODEL_TEXT_INIT = "Init";
        private const string MODEL_TEXT_CHANGE = "Change";

        private Root root;
        private ModelImpl model;
        private IObserver observer;

        [TestInitialize]
        public void SetUp()
        {
            root = StateTestUtils.SetUpRoot();
            model = ModelImpl.Create(root, MODEL_ID, MODEL_TEXT_INIT);
            StateTestUtils.Close(root);
            var mock = new Mock<IObserver>();
            observer = mock.Object;
            model.AddObserver(observer);
        }

        [TestMethod]
        public void TestModel()
        {
            Assert.AreEqual(MODEL_TEXT_INIT, model.ToText());
            model.ChangeText(MODEL_TEXT_CHANGE);
            StateTestUtils.Close(root);
            Assert.AreEqual(MODEL_TEXT_CHANGE, model.ToText());
            //verify(observer).update(MODEL_TEXT_CHANGE);
            Mock.Get(observer).Verify(observer => observer.Update(MODEL_TEXT_CHANGE), Times.Once());
        }
    }
}
